cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -115,6 +117,20 @@ public static LoggingContext getLoggingContext(Map<String, String> tags) {
                                        tags.get(ApplicationLoggingContext.TAG_RUN_ID),
                                        tags.get(ApplicationLoggingContext.TAG_INSTANCE_ID));
     } else if (tags.containsKey(WorkflowLoggingContext.TAG_WORKFLOW_ID)) {
+      if (tags.containsKey(WorkflowProgramLoggingContext.TAG_WORKFLOW_MAP_REDUCE_ID)) {
+        return new WorkflowProgramLoggingContext(namespaceId, applicationId,
+                                                 tags.get(WorkflowLoggingContext.TAG_WORKFLOW_ID),
+                                                 tags.get(ApplicationLoggingContext.TAG_RUN_ID), ProgramType.MAPREDUCE,
```

Comment:
@gokulavasan In order to collect the logs at the Workflow level we pass the runid of the Workflow. We also set the program name however. We don't need to include the runid of the program because when the query for the program run log is made, we derived the Workflow run id from program's run record and create appropriate WorkflowProgramLoggingContext.